src/national.py

```diff
@@ -1,34 +1,75 @@
-"""National API routes"""
+"""Fetch BMRS forecast data."""
 
 import os
+from datetime import datetime, timedelta
 from typing import List, Optional, Union
 
-import structlog
-from fastapi import APIRouter, Depends, HTTPException, Request, Security
-from fastapi_auth0 import Auth0User
-from nowcasting_datamodel.read.read import get_latest_forecast_for_gsps
+import pandas as pd
 from sqlalchemy.orm.session import Session
 
+import structlog
 from auth_utils import get_auth_implicit_scheme, get_user
 from cache import cache_response
 from database import (
     get_latest_forecast_values_for_a_specific_gsp_from_database,
     get_session,
     get_truth_values_for_a_specific_gsp_from_database,
 )
+from elexonpy.api.generation_forecast_api import GenerationForecastApi
+from elexonpy.api_client import ApiClient
+from fastapi import APIRouter, Depends, HTTPException, Query, Request, Security
+from fastapi_auth0 import Auth0User
+from nowcasting_datamodel.read.read import get_latest_forecast_for_gsps
 from pydantic_models import NationalForecast, NationalForecastValue, NationalYield
 from utils import N_CALLS_PER_HOUR, filter_forecast_values, format_datetime, format_plevels, limiter
 
 logger = structlog.stdlib.get_logger()
 
-
 adjust_limit = float(os.getenv("ADJUST_MW_LIMIT", 0.0))
 get_plevels = bool(os.getenv("GET_PLEVELS", True))
 
 router = APIRouter(
     tags=["National"],
 )
 
+# Initialize Elexon API client
+api_client = ApiClient()
+forecast_api = GenerationForecastApi(api_client)
+
+
+@router.get("/bmrs", summary="Get BMRS Forecast")
+def get_bmrs_forecast(
```

Comment:
sorry. Could we change `bmrs` to `elexon` too please. I think this is in a few places